app/Http/Controllers/LinkController.php

```diff
@@ -51,16 +51,23 @@ public function performRedirect(Request $request, $short_url, $secret_key=false)
         $link = Link::where('short_url', $short_url)
             ->first();
 
-        // Return 404 if link not found
+        // Return an error if the link does not exist
+        // or return a 404 if SETTING_REDIRECT_404 is set to true
         if ($link == null) {
-        	return abort(404);
+            if (env('SETTING_REDIRECT_404')) {
```

Comment:
If this setting is set, we want to redirect the user to the site's redirect URL. See the error handler for how 404s are handled.